src/Sarif/Autogenerated/ExternalPropertiesComparer.cs

```diff
@@ -38,13 +38,13 @@ public int Compare(ExternalProperties left, ExternalProperties right)
                 return compareResult;
             }
 
-            compareResult = string.Compare(left.Guid, right.Guid);
+            compareResult = left.Guid.CompareTo(right.Guid);
             if (compareResult != 0)
             {
                 return compareResult;
             }
 
-            compareResult = string.Compare(left.RunGuid, right.RunGuid);
+            compareResult = left.RunGuid.CompareTo(right.RunGuid);
```

Comment:
>left.RunGuid.CompareTo(right.RunGuid) [](http://example.com/codeflow?start=28&length=37)

This is where we would prefer Nullable.Compare<Guid>(left, right), if possible. Would this require an update to JSchema?

Reply:
Good catch. I have first updated my extension method to utilize the C# built-in method.

Nullable.Compare(left, right) will need to update the JSchema, it will need to be able to identify if a type is nullable, also struct not ref type and generate different code:
from:
`left.RunGuid.CompareTo(right.RunGuid);`
to:
`Nullable.Compare(left.RunGuid., right.RunGuid)`

also all those other nullable in the notyetautogenerated folder, in the JSchema we will not be able to know they are going to be replaced manually and we will read them as not nullable there.
By adding a extension it works with above.

let me know your instruction.